lib/fswatcher/fswatcher.go

```diff
@@ -108,14 +115,23 @@ func (watcher *FsWatcher) watchFilesystem() {
 	}
 }
 
-func (watcher *FsWatcher) newFsEvent(eventPath string) *FsEvent {
-	if isSubpath(eventPath, watcher.folderPath) {
+func (watcher *FsWatcher) newFsEvent(eventPath string) {
+	if len(watcher.fsEvents) == maxFiles {
+		watcher.debugf("Tracking too many events; dropping: %s\n", eventPath)
```

Comment:
Since this limit is meant to only be reached when there's too much going on to keep track of it's probably not the best time to order a full rescan and end up further increasing the load on the device. I think a better alternative is to ask for a full rescan when 'things have calmed down'. Possibly by setting an flag when the limit is hit and later when a batch of less than maxFiles events has been sent (or about to be sent) off to syncthing-core we can request a full scan.